singularity-eos/eos/modifiers/minimum_energy.hpp

```diff
@@ -0,0 +1,378 @@
+//------------------------------------------------------------------------------
+// © 2021-2024. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_MINIMUM_ENERGY_
+#define _SINGULARITY_EOS_EOS_MINIMUM_ENERGY_
+
+#include "stdio.h"
+#include <cstdlib>
+#include <iostream>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/base/eos_error.hpp>
+#include <singularity-eos/eos/eos_base.hpp>
+
+namespace singularity {
+
+using namespace eos_base;
+
+template <typename T>
+class MinimumEnergy : public EosBase<MinimumEnergy<T>> {
```

Comment:
I'm open to renaming this if someone has a more descriptive name

Reply:
Maybe `FlooredEnergy` or something?

Reply:
I like that better. I'll change it when I get back